### DISCUSSION TITLE
[@types/aws-lambda] property CloudFrontS3Origin.region should be conditional

### DISCUSSION THREAD

**Discussion by ransome** [DefinitelyTyped/DefinitelyTyped#68844] [Issues with a @types package]
In the following definition set:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/aws-lambda/common/cloudfront.d.ts

The property `CloudFrontS3Origin.region` should be conditionally required on whether `CloudFrontS3Origin.authMethod` is "origin-access-identity" or "none"

See docs here: https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/lambda-event-structure.html#:~:text=region%20(read/write,access%2Didentity.

Also below is an output of a test where a CloudFrontOrigin event was passed to a Lambda that console.log out the event:

`
                    "origin": {
                        "s3": {
                            "authMethod": "none",
                            "customHeaders": {},
                            "domainName": "rac-origin-test-s3.s3.eu-west-1.amazonaws.com",
                            "path": "/site"
                        }
                    },
`

Notice the "region" property is missing because `authMethod` is "none"

[a comment by an automated account is not shown]

**Comment by simonbuchan** [comment]
Nice catch! Might be tricky to avoid *any* breaking change, but it's worth it if the fix is easy enough.

**Comment by simonbuchan** [comment]
Let me know if #68849 looks good to you.

**Comment by ransome** [comment]
I'm not an expert for type definitions but what you did looks right to me.  Looks like it just got approved as well.  Thank you for the quick response!

**Comment by simonbuchan** [comment]
No problem, should be merging soon.
